File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
while True :
    length = list(map(int, input().split()))
    
    if length == [0, 0, 0] :
        break
    
    else : 
        if sorted(length)[-1] < sorted(length)[0] + sorted(length)[1] :
                if length.count(length[0]) == 3 :
                    print ("Equilateral")
            
                elif length[0] == length[1] or length[1] == length[2] or length[0] == length[2] :
                    print ("Isosceles")
        
                elif length[0] != length[1] and length[1] != length[2] and length[0] != length[2] :
                    print ("Scalene")
                    
        else : 
            print ("Invalid")
            

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.cc
#include <cstdio>
#include <iostream>
#include <map>
#include <string>
#include <algorithm>

int main() {
    while (true) {
        int i, j, k;
        scanf ("%d %d %d", &i, &j, &k);
        
        if (i == j && j == k && i == 0) {
            break;
        }
        
        int arr[] = {i, j, k};
        int size = sizeof(arr) / sizeof(*arr);
        
        int max = *std::max_element(arr, arr + size);
        int ind;
        
        // find index of max
        for (int l = 0; l < 3; l++) {
            if (max == arr[l]) {
                ind = l;
                break;
            }
        }
        
        int suma = 0;
        for (int l = 0; l < 3; l++) {
            if (ind == l) {
                continue;
            }
            
            suma += arr[l];
        }
        
        // printf ("%d %d\n", suma, max);
        
        if (suma <= max) {
            printf ("Invalid\n");
            continue;
        }
        
        if (i == j && j == k) {
            printf ("Equilateral\n");
        }
        
        else if (i == j || j == k || i == k) {
            printf ("Isosceles\n");
        }
        
        else if (i != j && j != k) {
            printf ("Scalene\n");
        }
    }
    
    return 0;
}
